shared/src/schema/user.schema.ts

```diff
@@ -0,0 +1,29 @@
+import { string } from 'zod'
+
+export const VALID_ALIAS_REGEX = /^(?=.{3,20}$)[a-zA-Z0-9]+(?:[_-][a-zA-Z0-9]+?)*$/
+
+const RESERVED_ALIAS = [
+  'admin',
+  'email',
+  'gast',
+  'gdd',
+  'gradido',
+  'guest',
+  'home',
+  'root',
+  'support',
+  'temp',
+  'tmp',
+  'tmp',
+  'user',
+  'usr',
+  'var',
+]
+
+export const aliasSchema = string()
+  .min(3, 'Given alias is too short')
```

Comment:
is "min(3..." equals <3 or <=3?
i simply ask to ensure the correct edges with the correct message.

Reply:
I have added a test for this, this is a good use case for a test, to checkout the borders. 
the tests show: 3 character is ok, 2 is not enough
```ts
it('2 characters is not ok', () => {
      expect(() => aliasSchema.parse('Bi')).toThrowError()
    })
    it('3 characters is ok', () => {
      expect(() => aliasSchema.parse('Bib')).not.toThrowError()
    })    
```
I had removed the test, because I don't think we need test for used libs that they work like they should. 
But we can test that they work like we expect.. if the semantic isn't clear enough, good point.